UnityProject/Assets/Scripts/Tilemaps/Behaviours/Layers/MetaDataLayer.cs

```diff
@@ -522,7 +579,7 @@ private void DistributeExcessToNearbyCells(ReagentMix excess, Vector3Int origin)
 				if (excess.Total <= 0) return;
 
 				// Skip if the neighbor is not passable
-				if (matrix.IsWallAt(neighbor, true)) continue;
+				if (matrix.IsWallAt(neighbor, true) || matrix.IsWindowAt(neighbor, true)) continue;
```

Comment:
This will still go through doors though

Reply:
It's intentional

Reply:
airlocks are sealed tho, they shouldn't let liquids pass

Reply:
fair point

Reply:
Updated to make it not pass through doors now.